watchtower/wtclient/client.go

```diff
@@ -765,6 +813,66 @@ func (c *TowerClient) nextSessionQueue() (*sessionQueue, error) {
 	return c.getOrInitActiveQueue(candidateSession, updates), nil
 }
 
+// handleChannelCloses listens for channel close events and marks channels as
+// closed in the DB.
+//
+// NOTE: This method MUST be run as a goroutine.
+func (c *TowerClient) handleChannelCloses(chanSub subscribe.Subscription) {
+	defer c.wg.Done()
+
+	c.log.Tracef("Starting channel close handler")
+	defer c.log.Tracef("Stopping channel close handler")
+
+	for {
+		select {
+		case update, ok := <-chanSub.Updates():
+			if !ok {
+				log.Debugf("channel notifier has exited")
+				return
+			}
+
+			// We only care about channel-close events.
+			event, ok := update.(channelnotifier.ClosedChannelEvent)
+			if !ok {
+				continue
+			}
+
+			chanID := lnwire.NewChanIDFromOutPoint(
+				&event.CloseSummary.ChanPoint,
+			)
+
+			// We only care about channels registered with the tower
+			// client.
+			c.backupMu.Lock()
+			_, ok = c.summaries[chanID]
```

Comment:
Q: could there be a race condition where we add the channel to the summaries later than we receive a close notification?

Reply:
even if that is possible, I dont think it would be an issue: if we first get a close-notification & then only register the channel then it is still the case that the channel is in fact closed. So no updates will be backed up for that channel. So no sessions would be affected by that channel closure. Do you agree?

Reply:
Agree and sounds good!